lib/vrl/stdlib/src/filter_array.rs

```diff
@@ -0,0 +1,141 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct FilterArray;
+
+impl Function for FilterArray {
+    fn identifier(&self) -> &'static str {
+        "filter_array"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "some item match",
+                source: r#"filter_array(["foobar", "bazqux"], r'foo')"#,
```

Comment:
I'm not sure if I would expect this to match if the regex doesn't match exhaustively. Maybe we can add an option for that behavior?

Reply:
`r'foo'` matches _foobar_, so I'd expect the current behaviour.

Reply:
I think this is fine. We don't provide an exhaustive option elsewhere, and you can always use `^foo$` if need be.